### ISSUE TITLE
Model validator retirement

### ISSUE THREAD

**Issue by plaidfinch** [penumbra-zone/penumbra#387]
Per @tarcieri, the Cosmos SDK contains a mechanism to allow validators to permanently retire without incurring slashing (as long as they have not committed any slashable offenses). As a follow-up to #375, we can support this as follows:

- [ ] Add another state, `Retired`, to the state model for validators. This state is not a state from which an unbonding period applies (undelegations should be immediate).
- [ ] Add special cases to the handling of validator state updates:
  - If an update causes a validator's sequence number to become `u32::MAX`, immediately transition that validator to the `Unbonding` state.
  - If a validator's sequence number is `u32::MAX` and it would otherwise move from the `Unbonding` state to the `Active` state, instead *do nothing* (it remains `Unbonding`).
  - If a validator's sequence number if `u32::MAX` and it is in the `Unbonding` state and the unbonding period has completed, move it to the `Retired` state (*not* the `Inactive` state).
- [ ] Add a new subcommand `pcli validator retire` which transmits a validator definition update setting the sequence number to `u32::MAX`.
- [ ] Update the status display in any other subcommands of `pcli` that display validator statuses so that a validator with sequence number `u32::MAX` in the `Unbonding` state is displayed as "retiring" and one in the `Retired` state is displayed as "retired".

The effect of the above is that if a validator runs `pcli validator retire`, their validator will begin to unbond, will never rejoin the active set regardless of their voting power, and at the end of the unbonding period will be permanently frozen with immediate undelegations available.

Design note: using `u32::MAX` as the special case to discriminate this state works nicely, because the consensus rule obligating monotonic increases of validator definition sequence number prevent any further changes to the validator definition after this point *anyhow*, which means this state is a terminal state for a validator already.

**Comment by hdevalence** [comment]
xref #375 ; we'll want to include this state transition in that writeup.

**Comment by zbuc** [comment]
> the consensus rule obligating monotonic increases of validator definition sequence number

FYI this isn't implemented as a *monotonic* increase, but it must be the highest of a set of increasing sequence numbers -- i.e. you could set your sequence number to `u32::MAX` manually if you wanted to.

**Comment by plaidfinch** [comment]
> it must be the highest of a set of increasing sequence numbers

Right, monotonic does not imply sequential, though it's a good point that what's meant here is _strongly monotonic_, i.e. "everywhere increasing", rather than weakly monotonic, i.e. "everywhere non-decreasing".
